ion-schema/src/test/kotlin/com/amazon/ionschema/model/ContinuousRangeTest.kt

```diff
@@ -0,0 +1,168 @@
+package com.amazon.ionschema.model
+
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.Arguments.arguments
+import org.junit.jupiter.params.provider.CsvSource
+import org.junit.jupiter.params.provider.MethodSource
+
+class ContinuousRangeTest {
+
+    @ParameterizedTest(name = "[constructor] ContinuousRange({0}, {1}) should be valid")
+    @MethodSource("validRanges")
+    fun `ContinuousRange should be valid`(min: ContinuousRange.Limit<Double>?, max: ContinuousRange.Limit<Double>?) {
+        ContinuousRange(min, max)
+    }
+    @ParameterizedTest(name = "[constructor] ContinuousRange({0}, {1}) should not be valid")
+    @MethodSource("invalidRanges")
+    fun `ContinuousRange should be invalid`(min: ContinuousRange.Limit<Double>?, max: ContinuousRange.Limit<Double>?) {
+        assertThrows<IllegalArgumentException> { ContinuousRange(min, max) }
+    }
+
+    @CsvSource(
+        // lower limit | high limit | value | is contained?
+        "CLOSED,  2.0, CLOSED,  4.0,   1.0, false",
```

Comment:
qq: why can we not just give kotlin lists for this? is it clearer as csv?

Reply:
If we use Kotlin lists, then I have to put the arguments in a `static` method in the `companion object`. Having them in CSV keeps them right by the test that actually uses the arguments. In some of the other tests, I've used the `@MethodSource` because the arguments were re-used across multiple tests or because the arguments were too complex or too many to hand-write as CSV.

Reply:
I like keeping them by the tests that uses the arguments. It wasn't (and still isn't) clear to me why that necessitates we use CsvSource except that apparently has some special mojo for this.

Reply:
I guess it's just that there is no other readily provided annotation for providing data in this way. odd.